### ISSUE TITLE
/createRoom results in a 403 + created room when mixing up the power levels

### ISSUE THREAD

**Issue by turt2live** [matrix-org/synapse#5579]
Creating a room with a name and power_level_content_override which doesn't list the creator as having power results in the correct "403 you don't have enough power" response, however the room is still created and synced down to the user. It should not be created.

**Comment by t3chguy** [comment]
Why 

> It should not be created

as opposed to defaulting to 100 if not provided

**Comment by t3chguy** [comment]
The spec in unclear in the granularity of the override, it does not say whether to do it on a depth=1 only or not:
> Overridden by the power_level_content_override parameter.

**Comment by turt2live** [comment]
I'd argue that https://matrix.org/docs/spec/client_server/r0.5.0#post-matrix-client-r0-createroom is pretty clear about the order of application.

**Comment by t3chguy** [comment]
Yes, but it doesn't say how to interpret `power_level_content_override`

Whether 

`power_level_content_override`:
```json
"users": {
    "@not_creator:server": 100
}
```

when overriding
```json
...
"users": {
    "@creator:server": 100
}
...
```

ends up including both of the users, or it merged on a depth of 1 only

**Comment by turt2live** [comment]
The intention is that the spec says it overrides the entire structure, not just at the root level. If that's unclear, please open a spec clarification.

**Comment by t3chguy** [comment]
Oh right, in synapse it uses a dict update,which is why it confused me. I'll fix it in synapse tomorrow

**Comment by t3chguy** [comment]
Corrected Synapse to match the above:

> The intention is that the spec says it overrides the entire structure, not just at the root level. If that's unclear, please open a spec clarification.

**Comment by richvdh** [comment]
> The intention is that the spec says it overrides the entire structure, not just at the root level. If that's unclear, please open a spec clarification.

The spec currently says of `power_level_content_override`: 

> This object is applied on top of the generated m.room.power_levels event content prior to it being sent to the room

"on top of" definitely suggests a merge, not an overwrite, to me, though I agree it's unclear how deep the merge should go. It feels like synapse kinda matches the spec at the moment, though.

@turt2live: are you sure about this?

**Comment by turt2live** [comment]
I was sure, but that quote leads me to believe that my life has been a lie. We should probably have a quick poll as the spec team and standardize on the winner.

**Comment by richvdh** [comment]
my main feeling on that is that the spec says what it says, and synapse seems to basically match it. There may be valid reasons to change it, but they probably need an MSC and I'm unconvinced they are worthwhile.

To return to the topic of this bug: there are probably hundreds of ways of specifying broken parameters to a `createRoom` call, all of which lead to a broken room being created... :/

**Comment by t3chguy** [comment]
> We should probably have a quick poll as the spec team and standardize on the winner.

Any update on this?

**Comment by turt2live** [comment]
@t3chguy no. https://github.com/matrix-org/synapse/issues/5579#issuecomment-513813024 is the last comment since then.

It was also a suggestion, not an action I plan on taking right now (hence "probably"). The core team is a bit involved with too many issues to worry about the subtleties of `/createRoom` at the moment. I would recommend at the very least opening a clarification PR to the spec as I said here: https://github.com/matrix-org/synapse/issues/5579#issuecomment-509029439

**Comment by t3chguy** [comment]
Fixed by #5633
